neqo-transport/src/send_stream.rs

```diff
@@ -306,6 +336,10 @@ impl RangeTracker {
             return;
         }
 
+        self.cached = None;
+        if self.extend_final_range(off, len as u64, state) {
+            return;
+        }
         let subranges = self.chunk_range_on_edges(off, len as u64, state);
 
         for (sub_off, sub_len, sub_state) in subranges {
```

Comment:
Another optimization, though maybe a small one:

Move `coalesce_acked_from_zero()` to `mark_as_acked()`.  Then make it faster.  (You said that you had a fix that improved that function, but I think we can do a whole lot better than that.)

Reply:
I don't like mark_as_acked(), since this isn't marking things.  cleanup_acked()?   The current name is long but descriptive

Reply:
Not sure that I'm following you.

The problem I'm trying to address is that coalesce_acked_from_zero is called when we mark things with any state, not just when we mark things as Acked.  We could move that function call to the place where we mark things as acked.  That is all.